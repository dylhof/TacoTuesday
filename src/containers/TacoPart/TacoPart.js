import React, { Component } from 'react';
import { connect } from 'react-redux';
import RecipeCard from '../../components/RecipeCard/RecipeCard';
import uuid from 'uuid/v4';

export class TacoPart extends Component {
  shiftTaco = (event) => {
    debugger
    const shiftDirection = event.target.classList.contains('left') ? -1 : 1
    const currentPosition = event.target.parentElement.children[1].scrollLeft;
    event.target.parentElement.childNodes[1].scrollLeft = currentPosition +(600 * shiftDirection)
  }

  render () {
    const part = this.props.tacoPart;
    const tacoPartRecipes = this.props[part].map(recipe => {
      return (
        <RecipeCard 
        key={uuid()} 
        tacoRecipe={recipe} styleName='tacoPartRecipes'
        listStyleName='tacoPartList'/>
      )
    });

    return (
      <div className='carouselDisplay'>
      <i className="fas fa-chevron-left fa-2x left" onClick={this.shiftTaco}></i>
        <div className='carousel'>
          {tacoPartRecipes}
        </div>
        <i className="fas fa-chevron-right fa-2x right" onClick={this.shiftTaco}></i>
      </div>
    )
  };
}

export const mapStateToProps = state => ({
  baseLayers: state.baseLayers,
  mixins: state.mixins,
  condiments: state.condiments,
  seasonings: state.seasonings,
  shells: state.shells
})

export default connect(mapStateToProps)(TacoPart);
