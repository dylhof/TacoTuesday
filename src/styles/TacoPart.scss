.carouselDisplay {
  display: flex;
  align-content: center;
  align-items: center;
  justify-content: center;
}

.carousel {
  display: flex;
  width: 90%;
  overflow: scroll;
}