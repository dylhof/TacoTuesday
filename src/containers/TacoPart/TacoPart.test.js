import React from 'react';
import { shallow } from 'enzyme';
import { TacoPart, mapStateToProps } from './TacoPart';


describe('TacoPart', () => {
  let wrapper;
  let mockBaseLayers
  let mockMixins
  let mockSeasonings
  let mockCondiments
  let mockShells

  beforeEach(() => {
    mockBaseLayers = [{slug: 'carnitas'}]
    mockMixins = [{slug: 'lettuce'}]
    mockSeasonings = [{slug: 'salt'}]
    mockCondiments = [{slug: 'salsa'}]
    mockShells = [{slug: 'bad_ass_tortillas'}, {slug: 'naan'}]
    wrapper = shallow(
      <TacoPart
      baseLayers={mockBaseLayers}
      mixins={mockMixins}
      condiments={mockCondiments}
      seasonings={mockSeasonings}
      shells={mockShells}
      tacoPart='shells'/> )
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it.skip('should shift the carousel', () => {
    const mockEvent = {event: {target: {classList: {0: 'left'}, parentElement:{children: {1: {scrollLeft: 0}}}}} }
    wrapper.instance().shiftTaco(mockEvent)
    expect(mockEvent.target.parentElement.children[1].scrollLeft).toBe(-600)
  });

  describe('mapStateToProps', () => {
    it('should return an object with all correct props', () => {
      //setup
      const mockState = {
        baseLayers: [{name: 'taco!'}],
        mixins: [{name: 'taco!'}],
        condiments: [{name: 'taco!'}],
        seasonings: [{name: 'taco!'}],
        shells: [{name: 'taco!'}],
        error: 'Sorry no tacos'
      }
      const expected = {
        baseLayers: [{name: 'taco!'}],
        mixins: [{name: 'taco!'}],
        condiments: [{name: 'taco!'}],
        seasonings: [{name: 'taco!'}],
        shells: [{name: 'taco!'}],
      }
      //execution
      const mappedProps = mapStateToProps(mockState)
      //expectation
      expect(mappedProps).toEqual(expected)
    })
  })
})