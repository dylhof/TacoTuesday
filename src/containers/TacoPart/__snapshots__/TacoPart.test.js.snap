// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TacoPart should match the snapshot 1`] = `
<div
  className="carouselDisplay"
>
  <i
    className="fas fa-chevron-left fa-2x left"
    onClick={[Function]}
  />
  <div
    className="carousel"
  >
    <RecipeCard
      key="1"
      listStyleName="tacoPartList"
      styleName="tacoPartRecipes"
      tacoRecipe={
        Object {
          "slug": "bad_ass_tortillas",
        }
      }
    />
    <RecipeCard
      key="1"
      listStyleName="tacoPartList"
      styleName="tacoPartRecipes"
      tacoRecipe={
        Object {
          "slug": "naan",
        }
      }
    />
  </div>
  <i
    className="fas fa-chevron-right fa-2x right"
    onClick={[Function]}
  />
</div>
`;
